detoks/src/main/java/nl/tudelft/trustchain/detoks/newcoin/AdminWallet.kt

```diff
@@ -4,40 +4,48 @@ import androidx.annotation.RequiresApi
 import nl.tudelft.trustchain.detoks.Token
 import nl.tudelft.trustchain.detoks.db.DbHelper
 
-data class AdminWallet(val tokens: ArrayList<Token>) {
+data class AdminWallet(
+    val publicKey: String
+) {
 
     companion object {
         private var wallet :AdminWallet? = null
-        private val dbHelper: DbHelper? = null
+        private var tokens : MutableList<Token>? = null
+        private var dbHelper: DbHelper? = null
         @RequiresApi(Build.VERSION_CODES.O)
         private fun create(context: Context): AdminWallet {
 //            val generator = KeyPairGenerator.getInstance("RSA")
 //            generator.initialize(2048)
 //            val keyPair = generator.generateKeyPair()
 
-            val dbHelper = DbHelper(context)
-            val tokens = dbHelper.getAllTokens()
+            dbHelper = DbHelper(context)
+            tokens = dbHelper!!.getAllTokens(admin = true)
+            wallet = AdminWallet("s")
```

Comment:
Hehe, public key = 's'? Was there any reason to not put the same public key again?

Reply:
The admin doesn't have a key. I had trouble initiating the wallet without an input so I left this here. Not a really clean fix so might need some revising.